receiver/opencensusreceiver/factory.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencensusreceiver
+
+import (
+	"context"
+
+	"github.com/census-instrumentation/opencensus-service/internal/configmodels"
+	"github.com/census-instrumentation/opencensus-service/internal/factories"
+
+	"github.com/census-instrumentation/opencensus-service/consumer"
+	"github.com/census-instrumentation/opencensus-service/receiver"
+)
+
+var _ = factories.RegisterReceiverFactory(&ReceiverFactory{})
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "opencensus"
+)
+
+// ReceiverFactory is the factory for receiver.
+type ReceiverFactory struct {
+}
+
+// Type gets the type of the Receiver config created by this factory.
+func (f *ReceiverFactory) Type() string {
+	return typeStr
+}
+
+// CustomUnmarshaler returns nil because we don't need custom unmarshaling for this config.
+func (f *ReceiverFactory) CustomUnmarshaler() factories.CustomUnmarshaler {
+	return nil
+}
+
+// CreateDefaultConfig creates the default configuration for receiver.
+func (f *ReceiverFactory) CreateDefaultConfig() configmodels.Receiver {
+	return &ConfigV2{
+		ReceiverSettings: configmodels.ReceiverSettings{
+			Endpoint: "127.0.0.1:55678",
```

Comment:
just a reminder for down the road: for the collector the default should be ":55678" (via the file defining the settings for the collector)

Reply:
I believe the proper way is to simply package the collector with its own default config file. So here we define some reasonable defaults which make sense both for Agent and Collector and any deviations should be reflected in the default config file that comes with the distribution. We will need to decide how exactly this packaging is done when we think about distribution and releases.